### ISSUE TITLE
Description of the Data files?

### ISSUE THREAD

**Issue by hani1986ye** [satoshifuku/Body_perception#13]
Daer Dr. Fukumori,

It is my first issue in Github, and I don't know if it will be posted correctly or not :)

First of all, thank you very much for inviting me to your project on Github.

I started to understand the analysis you did about the judgment (Data in the directory: ./Analysis/Data/[n]/Exp_[date]_judge.csv) by studying the python codes you created.

I decided to write codes in python and I tried to modify the write a code to do my task (plot 2D graphs that define the translational drift of both head and hand using the data in **./Analysis/Data/[n]/Exp_[date].csv**). 
But I got confused about the data in each column and I didn't find a description for them.

Can you give me a brief description of each data column especially the first column "s"? and why each column has more than 5000 rows

I also would like you to explain the meaning of some empty rows, rows with "**-1**", "**0**" and so on, and how often are they repeated in the data files?

I am attaching some screenshots here to explain what I am asking about.

![empty](https://user-images.githubusercontent.com/6098351/72439831-004b5280-37eb-11ea-88f7-5ae5880153d3.png)
![minusone](https://user-images.githubusercontent.com/6098351/72439847-08a38d80-37eb-11ea-8844-c4a1fe72483e.png)

Thank you.

**Comment by satoshifuku** [comment]
Mr. Hani

I'm sorry for replying late.

I think your confusion may be natural, because I did not explain that the data(.csv) includes three extra information.
In the disucussion, I explained only core structure of the data to disucuss simply.
I'm sorry that I do not send explations about a detail of the data.


First I'll explain the data's core strucure.
The data has 16 columns. 
Most left columns(index=s) is count of switching images between virtual hand and  monotone color
The next column(index=t) is elapsed time in one experimental trial.
The other 14 columns can be separate two groups which are head and hand.
Two groups has same data structure which consists of 7 columns about translation(x,y and z) and quaternion(q=w, x, y and z).

|s|t|XYZ|q_wxyz|XYZ|q_wxyz|
|:---|:---|:---|:---|:---|:---|
|---|---|---|---|---|---|
|---|---|---|---|---|---|


When you plot movements of head and hand in an experimetal trial to a graph , you MUST remove or ignore some rows based on three extra information.

**Comment by satoshifuku** [comment]
I'll explain the three extra information below.

First, you can find a blank row(b-row) in the data.
The b-row means thata an experimental trial just started.
Threfore, you should remove or ignore rows from the first row to a b-row.


|s|t|XYZ|q_wxyz|XYZ|q_wxyz|
|:---|:---|:---|:---|:---|:---|
|---|---|---|---|---|---|
| | | | | | |
|---|---|---|---|---|---|


![Inkedunnamed1_LI](https://user-images.githubusercontent.com/8034356/72734292-bee8e780-3bdc-11ea-86c6-ce044c372c49.jpg)

**Comment by satoshifuku** [comment]
Second, you can find some all zero's rows(Z-row).
The Z-row means visual stimulus just switched.
I guess that you have already been aware of that the number in the most right column counts up after a Z-row．
Therefore you MUST remove or ignore every  Z-rows.

|s|t|XYZ|q_wxyz|XYZ|q_wxyz|
|:---|:---|:---|:---|:---|:---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|


In addition, two consecutive Z-rows means that an experimental trial finished.
Therefore, you MUST remove or ignore rows from double Z-rows to the last row.

|s|t|XYZ|q_wxyz|XYZ|q_wxyz|
|:---|:---|:---|:---|:---|:---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|

**Comment by satoshifuku** [comment]
Last extra information is all -1's rows(M1-row).
The M1-row means a button of a controller was pushed.
Therefore, you MUST remove or ignore every M1 rows.

|s|Elaped time|T(xyz)|R(xyz)|T(xyz)|R(xyz)|
|:---|:---|:---|:---|:---|:---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|
|---|---|---|---|---|---|
|-1 |-1 |-1 |-1 |-1 |-1 |
|---|---|---|---|---|---|
|---|---|---|---|---|---|
|0|0 |0 |0 |0 |0 |
|---|---|---|---|---|---|

**Comment by hani1986ye** [comment]
Dr. Fukumori,
First, I am sorry for the late reply.

Thank you very much for your detailed explanation. 
Now I think the date's structure is clear.

I just still have a small question regarding the two groups of data for the hand and head. I understand from the values of X "height" that the first group is for the hand and the second is for the head. In my understanding, the zero-point "X=0" is the ground of the room; that means the hand is in a higher position than the head (for example the values X= 114 and X= 40 are for the head and the hand respectively). !!

I need to make sure if my understanding is correct or not, because your notation in the screenshot below says the opposite.

![SharedScreenshot](https://user-images.githubusercontent.com/6098351/73052260-0c5ca180-3ec8-11ea-8634-66b62dbc6a6d.jpg)

Thanks so much!

**Comment by satoshifuku** [comment]
> that means the hand is in a higher position than the head (for example the values X= 114 and X= 40 are for the head and the hand respectively). !!

You are understanding correctly!!
Participants lay on a bed and extend their hand to front of their chest.

**Comment by hani1986ye** [comment]
Dear Dr. Fukumori,

First of all, I am so sorry for the late response; I was a little sick and busy with my classes and research.

Second, I just finished a draft of the code to plot the range of movement of the participants' head and hand. 
I am not sure how to add my python file to the project, so I am attaching the file below:
[analysis_plot_h.zip](https://github.com/satoshifuku/Body_perception/files/4198260/analysis_plot_h.zip)

**Here I will try to summarize what I did:**
- First I tried to remove all unwanted rows manually, but I found that this way is not good, so I wrote a code to automatically remove the b-row, Z-rows, M1-rows, double Z-rows, and the other unrelated data according to your description. The cleaned data are stored in another directory "./Analysis/Data_cleaned/[n]/Exp_[date].csv", so the original files are kept without any change.

- Then I extracted the (X,Y,Z) information for each participant as well as for the whole experiments. These data are stored in another directory: "./Analysis/**result_h**/[n]/coordinates_[n].csv"

- After that, I used the X and Y coordinated to plot the range of movement. In this concern, I faced some issues regarding the practical meaning of the (X,Y,Z) coordinates.  At this time I considered Z as the height, so I didn't use it for the 2-D plot. I also used (X1,Y1,Z1) for the hand, and (X2,Y2,Z2) for the head. <<<- I need you comments if this is correct or not.

- Because the reference point (zero-point) is not clear for me, I normalized the data before the plotting to get a unified scale.

- I made very simple plots; I am not satisfied with these plots, and I think it should be changed, but I need your comments about the current figures before I change.

- In addition, I noticed that some data points might be incorrect can be considered outliers that alter the pattern of the plot.  <<< I also need your suggestion here to deal with these data points.

I am looking forward to receiving your comment.

Thanks so much.

**Comment by satoshifuku** [comment]
Never mind your late to replay.We are progressing :) 
Thank you for making a script to plot.

I added your script to the project in a new branch(feature).
See the following URL.
> https://github.com/satoshifuku/Body_perception/tree/feature/Analysis

Did you clone the project your local computer? i mean not download?
When you have already cloned, run the command to check your script was added.

> git checkout -b feature origin/feature

When you have not clone yet, i recommend you run "clone" because clone is the first step of a git workflow.
The following URL has good explanation  to workflow with the use of Github.
> https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/github-flow
> https://guides.github.com/introduction/flow/

**Comment by satoshifuku** [comment]
I opened new issue #15. Check it :)

**Comment by hani1986ye** [comment]
> Did you clone the project to your local computer? I mean not download?

No, I didn't clone the project because I didn't know how to clone :(

Now I downloaded GitHub Desktop, and I will learn how to clone the project.

Thanks for your recommendation.
